src/video-custom-controls.js

```diff
@@ -0,0 +1,603 @@
+        if (!size) {
+          size = progressBar./*OK*/getBoundingClientRect();
+        }
+        const left = size.left;
+        const total = size.width;
+        const newPercent = Math.min(100,
+            Math.max(0, 100 * (e.clientX - left) / total)
+        );
+        scrubberDragging = scrubberTouched;
+        if (scrubberDragging) {
+          st.setStyles(scrubber, {
+            'left': st.percent(newPercent),
+          });
+          st.setStyles(currentBar, {
+            'width': st.percent(newPercent),
+          });
+        }
+      });
+    });
+
+    ['mouseup', 'touchend'].forEach(event => {
+      listen(doc, event, () => {
+        scrubberTouched = false;
+        scrubberDragging = false;
+      });
+    });
+
+    // Update progress bar
+    const updateProgress = () => {
+      const current = this.entry_.video.getCurrentTime() || 0;
+      const total = this.entry_.video.getDuration() || 0;
+      const percent = total ? Math.floor(current * (100 / total)) : 0;
+      st.setStyles(currentBar, {
+        'width': st.percent(percent),
+      });
+      st.setStyles(scrubber, {
+        'left': st.percent(percent),
+      });
+    };
+
+    [VideoEvents.TIME_UPDATE, VideoEvents.LOAD].forEach(e => {
+      listen(this.entry_.video.element, e, updateProgress.bind(this));
+    });
+
+    return progressBar;
+  }
+
+  createIcon_(name) {
+    const doc = this.ampdoc_.win.document;
+    const icon = doc.createElement('amp-custom-ctrls-icon');
+    icon.className = 'amp-custom-ctrls-icon-' + name;
+    return icon;
+  }
+
+  changeIcon_(icon, name) {
+    icon.className = 'amp-custom-ctrls-icon-' + name;
+  }
+
+  /**
+   * Creates a button element from the button's name
+   * @param {string} btn
+   * @param {?Element|string} opt_loadingElement
+   * @return {!Element}
+   * @private
+   */
+  elementFromButton_(btn, opt_loadingElement = null) {
+    const doc = this.ampdoc_.win.document;
+    switch (btn) {
+      case 'play':
+        return this.createPlayPauseBtn_(opt_loadingElement);
+      case 'time':
+        return this.createProgressTime_();
+      case 'spacer':
+        return this.createSpacer_();
+      case 'volume':
+        return this.createVolumeControls_();
+      case 'fullscreen':
+        return this.createFullscreenBtn_();
+      default:
+        return doc.createElement('span');
+    };
+  }
+
+  /**
+   * Create the custom controls shim and insert it inside the video
+   * @param {Array<string>} mainCtrls List of orderd controls (main bar)
+   * @param {Array<string>} miniCtrls List of orderd controls (minimized ctrls)
+   * @param {string} floating Name of the primary floating button (usually play)
+   * @private
+   */
+  createCustomControls_(mainCtrls, miniCtrls, floating) {
+    // Set up controls
+    const doc = this.ampdoc_.win.document;
+    this.ctrlContainer_ = doc.createElement('amp-custom-ctrls');
+    this.ctrlContainer_.classList.toggle('light', !this.darkSkin_);
```

Comment:
Since it's not a user-entered value, we should definitely prefix and make it pretty explicit too. E.g. `amp-custom-controls-skin-light` or such. There's a very non-trivial probability that the existing stylesheets already have `.light` and `.dark` classes and they mean something very different.